ui-tests/tst/software/aws/toolkits/jetbrains/uitests/s3/S3BrowserTest.kt

```diff
@@ -0,0 +1,200 @@
+import com.intellij.remoterobot.stepsProcessing.step
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.AfterAll
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.TestInstance
+import org.junit.jupiter.api.TestInstance.Lifecycle
+import org.junit.jupiter.api.io.TempDir
+import software.aws.toolkits.jetbrains.uitests.CoreTest
+import software.aws.toolkits.jetbrains.uitests.extensions.uiTest
+import software.aws.toolkits.jetbrains.uitests.fixtures.JTreeFixture
+import software.aws.toolkits.jetbrains.uitests.fixtures.awsExplorer
+import software.aws.toolkits.jetbrains.uitests.fixtures.fillSingleTextField
+import software.aws.toolkits.jetbrains.uitests.fixtures.findAndClick
+import software.aws.toolkits.jetbrains.uitests.fixtures.idea
+import software.aws.toolkits.jetbrains.uitests.fixtures.pressDelete
+import software.aws.toolkits.jetbrains.uitests.fixtures.pressOk
+import software.aws.toolkits.jetbrains.uitests.fixtures.welcomeFrame
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.time.Duration
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.util.UUID
+
+@TestInstance(Lifecycle.PER_CLASS)
+class S3BrowserTest {
+    private val testDataPath: Path = Paths.get(System.getProperty("testDataPath"))
+
+    private val profile = "default"
+    private val credential = "Profile:$profile"
+    private val region = "Oregon (us-west-2)"
+    private val date = LocalDate.now().format(DateTimeFormatter.BASIC_ISO_DATE)
+    private val bucket = "uitest-$date-${UUID.randomUUID()}"
+    private val folder = UUID.randomUUID().toString()
+
+    private val S3 = "S3"
+    private val createBucketText = "Create S3 Bucket"
+    private val deleteBucketText = "Delete S3 Bucket"
+    private val upload = "Upload..."
+    private val newFolder = "New folder..."
+    private val rename = "Rename..."
+    private val delete = "Delete..."
+
+    private val jsonFile = "hello.json"
+    private val jsonFile2 = "hello2.json"
+    private val newJsonName = "helloooooooooo.json"
+
+    @TempDir
+    lateinit var tempDir: Path
+
+    @Test
+    @CoreTest
+    fun testS3Browser() = uiTest {
+        welcomeFrame {
+            newProject(tempDir)
+        }
+        idea {
+            waitForBackgroundTasks()
+            setCredentials(credential, region)
+            showAwsExplorer()
+        }
+        idea {
+            step("Create bucket named $bucket") {
+                awsExplorer {
+                    openExplorerActionMenu(S3)
+                }
+                find<ComponentFixture>(byXpath("//div[@text='$createBucketText']")).click()
+                find<JTextFieldFixture>(byXpath("//div[@class='JTextField']"), Duration.ofSeconds(5)).text = bucket
+                find<ComponentFixture>(byXpath("//div[@text='Create']")).click()
+            }
+
+            // Wait for the bucket to be created
+            Thread.sleep(10000)
```

Comment:
We should just pull in the SDK for these checks so that we dont aimlessly slow down our tests. Also would help setting state for tests too

Reply:
This is how we did it in the old tests, but this isn't as simple as pulling in the SDK because in CI the credentials are in a nonstandard location :(. I'll see if there is a quick way to add it